pkg/neotest/basic.go

```diff
@@ -0,0 +1,261 @@
+	"github.com/nspcc-dev/neo-go/pkg/io"
+	"github.com/nspcc-dev/neo-go/pkg/smartcontract/callflag"
+	"github.com/nspcc-dev/neo-go/pkg/smartcontract/trigger"
+	"github.com/nspcc-dev/neo-go/pkg/util"
+	"github.com/nspcc-dev/neo-go/pkg/vm"
+	"github.com/nspcc-dev/neo-go/pkg/vm/emit"
+	"github.com/nspcc-dev/neo-go/pkg/vm/stackitem"
+	"github.com/nspcc-dev/neo-go/pkg/wallet"
+	"github.com/stretchr/testify/require"
+)
+
+// Executor is a wrapper over chain state.
+type Executor struct {
+	Chain         blockchainer.Blockchainer
+	Validator     Signer
+	Committee     Signer
+	CommitteeHash util.Uint160
+	Contracts     map[string]*Contract
+}
+
+// NewExecutor creates new executor instance from provided blockchain and committee.
+func NewExecutor(t *testing.T, bc blockchainer.Blockchainer, validator, committee Signer) *Executor {
+	checkMultiSigner(t, validator)
+	checkMultiSigner(t, committee)
+
+	return &Executor{
+		Chain:         bc,
+		Validator:     validator,
+		Committee:     committee,
+		CommitteeHash: committee.ScriptHash(),
+		Contracts:     make(map[string]*Contract),
+	}
+}
+
+// TopBlock returns block with the highest index.
+func (e *Executor) TopBlock(t *testing.T) *block.Block {
+	b, err := e.Chain.GetBlock(e.Chain.GetHeaderHash(int(e.Chain.BlockHeight())))
+	require.NoError(t, err)
+	return b
+}
+
+// NativeHash returns native contract hash by name.
+func (e *Executor) NativeHash(t *testing.T, name string) util.Uint160 {
+	h, err := e.Chain.GetNativeContractScriptHash(name)
+	require.NoError(t, err)
+	return h
+}
+
+// NewUnsignedTx creates new unsigned transaction which invokes method of contract with hash.
+func (e *Executor) NewUnsignedTx(t *testing.T, hash util.Uint160, method string, args ...interface{}) *transaction.Transaction {
+	w := io.NewBufBinWriter()
+	emit.AppCall(w.BinWriter, hash, method, callflag.All, args...)
+	require.NoError(t, w.Err)
+
+	script := w.Bytes()
+	tx := transaction.New(script, 0)
+	tx.Nonce = nonce()
+	tx.ValidUntilBlock = e.Chain.BlockHeight() + 1
+	return tx
+}
+
+// NewTx creates new transaction which invokes contract method.
+// Transaction is signed with signer.
+func (e *Executor) NewTx(t *testing.T, signers []Signer,
+	hash util.Uint160, method string, args ...interface{}) *transaction.Transaction {
+	tx := e.NewUnsignedTx(t, hash, method, args...)
+	return e.SignTx(t, tx, -1, signers...)
+}
+
+// SignTx signs a transaction using provided signer.
+func (e *Executor) SignTx(t *testing.T, tx *transaction.Transaction, sysFee int64, signers ...Signer) *transaction.Transaction {
+	for _, acc := range signers {
+		tx.Signers = append(tx.Signers, transaction.Signer{
+			Account: acc.ScriptHash(),
+			Scopes:  transaction.Global,
+		})
+		addNetworkFee(e.Chain, tx, acc.Script())
+	}
+	addSystemFee(e.Chain, tx, sysFee)
+
+	for _, acc := range signers {
+		require.NoError(t, acc.SignTx(e.Chain.GetConfig().Magic, tx))
+	}
+	return tx
+}
+
+// NewAccount creates new account and transfers 100.0 GAS to it.
```

Comment:
... returns new Signer associated with new single-key account holding 100 GAS. This method advances the chain by one block with transfer transaction (100 GAS from committee to the new account).

An additional block can be important for some tests, so it'd be nice to mention.